### DISCUSSION TITLE
Virtual Boy

### DISCUSSION THREAD

**Discussion by ionicbox** [sanni/cartreader#353] [Ideas & Suggestions]
Any chance of adding support for the Virtual Boy

**Comment by sanni** [comment]
https://mellottsvrpage.com/index.php/vb-cart-programmer-quick-start-guide/ seems to use my Flashrom Writer 16bit functions out of flash.ino to dump Virtual Boy. I'm not sure if he did any changes to the code or if it just works as-is.

You could start by wiring up the Virtual Boy cart slot like the 16bit flash pinout in [pinout.ods](https://github.com/sanni/cartreader/blob/master/pinout.ods) combined with the Virtual Boy pinout from [here](https://www.projectvb.com/vb/tech/cartpinout.html) and see what happens when dumping using the existing 29F1610 flashrom functions with slight changes.

I don't think that the Virtual Boy uses any mappers so it could just work.

**Reply by 7ank0v1c** [reply]
![5CBE664C-7D32-4B6B-ADEE-B809F269899A](https://user-images.githubusercontent.com/40895945/212220016-72baf82a-bce9-4a2c-8ca1-76446b490f13.jpeg)

![DCD545F1-0C9D-4C16-A76B-3A9D6DDB24A7](https://user-images.githubusercontent.com/40895945/212220029-481295f8-a60b-45e9-9814-720167492cb8.jpeg)

![EA93F6D5-96A4-49CD-A326-93ED1BD6B525](https://user-images.githubusercontent.com/40895945/212220032-56d55e14-d5b3-471b-aabb-c638686db29c.jpeg)

![719851D5-E065-4324-91C5-9D3233A1A5FE](https://user-images.githubusercontent.com/40895945/212220036-66de5132-1161-4a4d-83d0-6fd641ef2e1f.jpeg)

This was really tough for me. Took me close to four hours. Almost gave up multiple times, however, I did it 🙌 and it works (even better 🙌)

Thanks guys for making this adapter available

**Reply by 7ank0v1c** [reply]
![976B3D5F-3EF1-49D0-9B87-522013E39EEB](https://user-images.githubusercontent.com/40895945/212220978-b37bd184-5258-4545-ba59-276e5965cbfe.jpeg)

This guide piece also acts as a good protection against bending the pins during storage
